### PR TITLE
Add Unit Tests for shared_utils.py

### PR DESCRIPTION
**Pull request by TraXIcoN**
This PR adds unit tests for the shared_utils.py module. It covers encoding utilities, bit manipulation, argument handling, ISA type validation, string manipulation, and instruction processing. These tests will help ensure the module works correctly and is easier to maintain.

### CODE REVIEW

**Comment by IIITM-Jay** [comment]
@TraXIcoN I have fixed the issue and removed the label `do-not-merge`. Will review it soon.

Looping @aswaterman and @rpsene also.

Thanks @TraXIcoN for this PR and adding test cases for `shared` module and its associated methods!

**Comment by IIITM-Jay** [comment]
@TraXIcoN LGTM. 

Definitely it contains well organized test suites comprising both **_positive and negative test cases_**. Once you resolved with conflicts, comment here so that it could get merged.

Let me know, if you run into an issue while fixing `pyright` things, will fix those.

File: test.py [hunk 1]
```diff
@@ -2,41 +2,238 @@
 
 import logging
 import unittest
+from unittest.mock import Mock, mock_open, patch
```
Comment (IIITM-Jay) [inline test.py:5]: @TraXIcoN I think, the `mock_open` is not being used anywhere in the test suites. Could you verify it once and remove that if not being used.

**Review by IIITM-Jay — APPROVED** [review]
@TraXIcoN  I have resolved the conflicts and fixed the `pyright` issues. Actually, the `pre-commit` tool for this repo is now also includes the `pyright` hooks. So, some changes were needed for test suites too, I have done the necessary changes accordingly.

@aswaterman requesting your feedback and an overview for this PR, and then I think this PR is good to be merged.

Thanks once again @TraXIcoN for the tests and @aswaterman for your extreme support!

**Review by aswaterman — APPROVED** [review]
Yes, I'm happy with this.  Thanks, @TraXIcoN.

**Pull request merged**